File: api/src/Domain/Billing/Entities/SubscriptionPlan.cs
using System;
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Billing {
    public class SubscriptionPlan : Aggregate<SubscriptionPlan>, IBillingEntity {
        public string Name { get; set; }
        public string? Description { get; set; }
        public BillingReference BillingReference { get; }
        public List<SubscriptionPlanPrice> Prices { get; set; }

        public SubscriptionPlan(string name, string? description, BillingReference billingReference, List<SubscriptionPlanPrice> prices) {
            Name = name;
            Description = description;
            BillingReference = billingReference;
            Prices = prices;
        }
    }
}

File: app/src/api/billing/data-transfer-objects/subscription-plan.ts
import { SubscriptionPlanPrice } from '@/api/billing/data-transfer-objects/subscription-plan-price';

/**
 * A plan that can be used to create subscriptions. A plan can hold multiple prices.
 * (EX one for monthly, one for yearly)
 */
export class SubscriptionPlan {
    constructor(
        public id: string,
        public name: string,
        public description: string | null,
        public prices: SubscriptionPlanPrice[]
    ) {}
}
